store/sqlstore/post_store.go

```diff
@@ -712,6 +713,51 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, numPosts
 	})
 }
 
+func (s *SqlPostStore) GetPostBefore(channelId, postId string) store.StoreChannel {
+	return s.getPostAround(channelId, postId, true)
+}
+
+func (s *SqlPostStore) GetPostAfter(channelId, postId string) store.StoreChannel {
+	return s.getPostAround(channelId, postId, false)
+}
+
+func (s *SqlPostStore) getPostAround(channelId string, postId string, before bool) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		var direction string
+		var sort string
+		if before {
+			direction = "<"
+			sort = "DESC"
+		} else {
+			direction = ">"
+			sort = "ASC"
+		}
+
+		var post *model.Post
+		err := s.GetReplica().SelectOne(
+			&post,
+			`(SELECT
+			    *
+			FROM
+			    Posts
+			WHERE
+				(CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
```

Comment:
I think the nested query here won't perform well. Instead, we could pass the `post.CreateAt` as an argument of this function instead of the `post.Id`